### ISSUE TITLE
packetfabric_cs_<aws/azure/google>_<hosted/dedicated>_connection delete aren't working

### ISSUE THREAD

**Issue by rjouhann** [PacketFabric/terraform-provider-packetfabric#83]
While testing the `packetfabric_cs_azure_hosted_connection` resource, I found out the creation works now after this fix https://github.com/PacketFabric/terraform-provider-packetfabric/pull/80 but the deletion is not working.

There are no errors but the Azure Hosted Connection doesn't get deleted. And it is the same for `packetfabric_cs_aws_hosted_connection` as well as `packetfabric_cs_google_hosted_connection`.

See output of the terraform destroy:
```
  # packetfabric_cs_azure_hosted_connection.pf_cs_conn1 will be destroyed
  - resource "packetfabric_cs_azure_hosted_connection" "pf_cs_conn1" {
      - account_uuid      = "24a0aef3-2784-48dc-9a8e-52d44d6da646" -> null
      - azure_service_key = (sensitive) -> null
      - description       = "demo-pf-azure-noble-cattle" -> null
      - id                = "172ccf44-fea5-4121-9816-2d0e60d678b2" -> null
      - port              = "PF-AP-WDC1-1726464" -> null
      - speed             = "50Mbps" -> null
      - vlan_private      = 100 -> null
    }

```

Debug logs:
```
packetfabric_cs_azure_hosted_connection.pf_cs_conn1: Destroying... [id=172ccf44-fea5-4121-9816-2d0e60d678b2]
packetfabric_cs_azure_hosted_connection.pf_cs_conn1: Destruction complete after 0s
azurerm_subnet.subnet_1: Destroying... [id=/subscriptions/d2a01bfc-a1dc-4af0-8600-ff7c8dd7bb5a/resourceGroups/demo-pf-azure-noble-cattle/providers/Microsoft.Network/virtualNetworks/demo-pf-azure-noble-cattle-vnet1/subnets/demo-pf-azure-noble-cattle-subnet1]
azurerm_subnet.subnet_gw: Destroying... [id=/subscriptions/d2a01bfc-a1dc-4af0-8600-ff7c8dd7bb5a/resourceGroups/demo-pf-azure-noble-cattle/providers/Microsoft.Network/virtualNetworks/demo-pf-azure-noble-cattle-vnet1/subnets/GatewaySubnet]
azurerm_subnet.subnet_1: Still destroying... [id=/subscriptions/d2a01bfc-a1dc-4af0-8600-...ets/demo-pf-azure-noble-cattle-subnet1, 10s elapsed]
azurerm_subnet.subnet_gw: Still destroying... [id=/subscriptions/d2a01bfc-a1dc-4af0-8600-...ble-cattle-vnet1/subnets/GatewaySubnet, 10s elapsed]
azurerm_subnet.subnet_gw: Destruction complete after 11s
azurerm_subnet.subnet_1: Still destroying... [id=/subscriptions/d2a01bfc-a1dc-4af0-8600-...ets/demo-pf-azure-noble-cattle-subnet1, 20s elapsed]
azurerm_subnet.subnet_1: Destruction complete after 22s
```

Could you double check you are using DEL API from [/v2/services/cloud/{cloud_circuit_id}](https://docs.packetfabric.com/api/v2/swagger/#/Cloud%20Services%20-%20Azure/cloud_connection_delete) for `packetfabric_cs_azure_hosted_connection`

This should be the same DELETE API for all cloud connections:

- packetfabric_cs_aws_hosted_connection
- packetfabric_cs_aws_dedicated_connection
- packetfabric_cs_google_hosted_connection
- packetfabric_cs_google_hosted_connection
- packetfabric_cs_azure_hosted_connection
- packetfabric_cs_azure_dedicated_connection

All should be using DEL `/v2/services/cloud/{cloud_circuit_id}`

The `packetfabric_backbone_virtual_circuit` delete is using DEL `/v2/services/{cloud_circuit_id}`. I think when you fixed https://github.com/PacketFabric/terraform-provider-packetfabric/issues/70, you broke all the delete for all hosted, dedicated and marketplace resource.

Essentially as explained in the other issue:

You use [this one](https://docs.packetfabric.com/api/v2/redoc/#tag/Cloud-Services-Azure/operation/cloud_connection_delete) `DEL /v2/services/cloud/{cloud_circuit_id}` for all cloud_services deletes:

```
packetfabric_cs_aws_hosted_connection
packetfabric_cs_aws_dedicated_connection

packetfabric_cs_google_hosted_connection
packetfabric_cs_google_dedicated_connection

packetfabric_cs_azure_hosted_connection
packetfabric_cs_azure_dedicated_connection
```

And [this one](https://docs.packetfabric.com/api/v2/swagger/#/Services/delete_service) `DEL /v2/services/{vc_circuit_id}` for `packetfabric_backbone_virtual_circuit` (don't touch this one, it's working now)

**Comment by rjouhann** [comment]
@pontovinte I see you are using the `id` attribute to delete the hosted connection while we should be using the `cloud_circuit_id` to delete the connection (hosted and dedicated). For hosted Marketplace connection, it is a little bit more complicated (see https://github.com/PacketFabric/terraform-provider-packetfabric/issues/91).

See below example TF output the `id` used instead of `cloud_circuit_id` is used which should look like something like `PF-CC-WDC-NYC-1726496-PF`.
```
packetfabric_cs_aws_hosted_connection.cs_conn1: Destroying... [id=demo-romain-maximum-crane]
packetfabric_cs_azure_hosted_connection.cs_conn1: Destroying... [id=172ccf44-fea5-4121-9816-2d0e60d678b2]
packetfabric_cs_google_hosted_connection.cs_conn1: Destroying... [id=b2fd0cfa-6e7d-45e1-ab24-0ece6b4ea3c9]
```

Fortunately, we can find the `cloud_circuit_id` for each cases:

For all hosted connection DELETE: 
- **packetfabric_cs_aws_hosted_connection**
- **packetfabric_cs_azure_hosted_connection**
- **packetfabric_cs_google_hosted_connection**

The API to create the connection returns `cloud_circuit_id` so we can use it for the DELETE.

Example Response:
POST /v2/services/cloud/hosted/google (same for aws and azure)
```
{
   ...
   "cloud_circuit_id":"PF-CC-ATL-SFO-1744245-PF",
   ...
}
```

For all dedicated connection DELETE: 
- **packetfabric_cs_aws_dedicated_connection**
- **packetfabric_cs_azure_dedicated_connection**
- **packetfabric_cs_google_dedicated_connection**

The API creating the connection returns a `null` value for `cloud_circuit_id`. But you can use the `uuid` returned in the response and look for the `cloud_circuit_id` using GET /v2/services/cloud/connections/dedicated and the `uuid`

Example Response:
POST /v2/services/cloud/dedicated/google
```
{
   "uuid":"e5da76ee-15ea-4408-be36-fc0a22097e16",
   ...
   "cloud_circuit_id":null,
   ...
   "state":"requested",
   ...
   "speed":"10Gbps"
}
```
Then do the GET and look for the `cloud_circuit_id` using the `uuid`.
GET /v2/services/cloud/connections/dedicated
```
[
   {
      "uuid":"e5da76ee-15ea-4408-be36-fc0a22097e16",
      ...
      "cloud_circuit_id":"PF-AE-OMA1-1744242",
      ...
      "state":"active",
      ...
      "speed":"10Gbps"
   }
]
```

**Comment by rjouhann** [comment]
@pontovinte see your PR https://github.com/PacketFabric/terraform-provider-packetfabric/pull/98 for hosted marketplace. But as mentioned in my previous post, for hosted Marketplace connection, it is a little bit more complicated (see https://github.com/PacketFabric/terraform-provider-packetfabric/issues/91) so we should address it in v0.2.1.

This issue is ONLY to fix both hosted and dedicated cloud resource delete for ALL clouds (AWS/Google/Azure).

**Comment by rjouhann** [comment]
Verified in [feature/google-service](https://github.com/PacketFabric/terraform-provider-packetfabric/tree/feature/google-service) branch.

- `packetfabric_cs_google_dedicated_connection`: creation NOK (same problem as before, the create never end)
- `packetfabric_cs_azure_dedicated_connection`: creation/deletion OK
- `packetfabric_cs_aws_dedicated_connection`: creation NOK (same problem as before, the create never end)
- `packetfabric_cs_aws_hosted_connection`: creation/deletion OK
- `packetfabric_cs_azure_hosted_connection`: create NOK (see error below
- `packetfabric_cs_google_hosted_connection`: creation/deletion OK

packetfabric_cs_azure_hosted_connection
```
│ Error: json: cannot unmarshal number into Go struct field Components.components.vc_id_private of type string
│ 
│   with packetfabric_cs_azure_hosted_connection.cs_conn1_hosted_azure,
│   on main.tf line 359, in resource "packetfabric_cs_azure_hosted_connection" "cs_conn1_hosted_azure":
│  359: resource "packetfabric_cs_azure_hosted_connection" "cs_conn1_hosted_azure" {
│ 
```

**Comment by rjouhann** [comment]
Verified in [feature/google-service](https://github.com/PacketFabric/terraform-provider-packetfabric/tree/feature/google-service) branch again (I deleted the entire folder, did a git clone, switch to the `feature/google-service` branch, then compile the local provider)

- `packetfabric_cs_google_dedicated_connection`: creation/deletion OK
- `packetfabric_cs_azure_dedicated_connection`: creation/deletion OK
- `packetfabric_cs_aws_dedicated_connection`: creation/deletion OK
- `packetfabric_cs_google_hosted_connection`: creation/deletion OK
- `packetfabric_cs_azure_hosted_connection`: creation/deletion OK
- `packetfabric_cs_aws_hosted_connection`: creation/deletion OK

However, I found another problem with dedicated connection, if the user puts something wrong in the PoP, Zone and the port cannot be provisioned, the resource will go forever and never return an error back, user will have to wait to hit the timeout :(

I have created an issue for this, very easy to repro, just pick a PoP and a Zone that doesn't exist. This works fine for hosted connection btw where the error is correctly reported.

I have opened another issue to handle errors better in v0.2.1.
https://github.com/PacketFabric/terraform-provider-packetfabric/issues/105
